### DISCUSSION TITLE
Display Env Variable in Right Status

### DISCUSSION THREAD

**Discussion by rrpolanco** [wezterm/wezterm#1321] [Q&A]
I'm trying to display the value of an environment variable, that I set on demand in the shell, via `wezterm.run_child_process()`  in the right status bar via `window:set_right_status()`. However, I've noticed that the envar I set is not known by the child process. Is there a way to tell the child process the env info of my current terminal session?

**Comment by bew** [comment]
Can you give some code of what you're trying to do?
Where are you setting the env variable?

Note that if it's in a shell running in the terminal you won't be able to get the env var, you can only access env vars of your current or parents processes, not your childs.

**Reply by rrpolanco** [reply]
Hi @bew 

I'm essentially trying to display my kubernetes context in the status bar. I'm attempting to achieve this by first calling an external program to get the current context and then putting the output of the command into the right-status.

Something like this:
```lua
local success, stdout, stderr = wezterm.run_child_process({"kubectx", "-c"});

<...>

window:set_right_status(stdout);
```

The issue is that I dynamically set the `KUBECONFIG`  envar in my running shell, which `kubectx` uses to give back the correct context, and the child process doesn't know about `KUBECONFIG` being set. I was just wondering if there was a way to probe the current running shell's envars or perhaps pass our own envars to the child_process.

**Reply by wez** [reply]
I'd suggest looking at *user vars*: https://wezfurlong.org/wezterm/config/lua/pane/get_user_vars.html
The idea is that your shell would set a user var to capture KUBECONFIG:

```bash
# Usage: set_user_var KEY VALUE
function set_user_var() {
   printf "\033]1337;SetUserVar=%s=%s\007" $1 `echo -n $2 | base64`
}

set_user_var KUBECONFIG "$KUBECONFIG"
```

and then your wezterm config could retrieve that and run something like:

```lua
  run_child_process({"env", "KUBECONFIG="..pane:get_user_vars().KUBECONFIG, "kubectx", "-c"})
```

**Reply by rrpolanco** [reply]
I was able to get my kubeconfig info into the status bar per your suggestion @wez. Thanks and keep up the great work!
